models/dit/index.js

```diff
@@ -0,0 +1,398 @@
+        [`${ditType}`]: dit,
+        tag: { tagname }
+      }));
+      return dit;
+    });
+  }
+
+  /**
+   * Read dits with tags
+   * @param {string[]} tagnames - list of tagnames to search with
+   * @param {integer} offset - pagination offset
+   * @param {integer} limit - pagination limit
+   * @returns {Promise<Dit[]>} - list of found dits
+   */
+  static async withTags(ditType, tagnames, { offset, limit }) {
+    const ditTags = ditType + 'Tags';
+    const query = `
+      // find the provided tags
+      FOR t IN tags FILTER t.tagname IN @tagnames
+        SORT t.tagname
+        LET tg = KEEP(t, 'tagname')
+        // find the related dits
+        FOR i IN 1..1 ANY t INBOUND @@ditTags
+          // collect tags of each dit together
+          COLLECT ${ditType}=i INTO collected KEEP tg
+          // sort dits by amount of matched tags, and from oldest
+          SORT LENGTH(collected) DESC, ${ditType}.created ASC
+          // read and format creator
+          LET c = (DOCUMENT(${ditType}.creator))
+          LET creator = MERGE(KEEP(c, 'username'), c.profile)
+          // format for output
+          LET ${ditType}Out = MERGE(KEEP(${ditType}, 'title', 'detail', 'created'), { id: ${ditType}._key}, { creator })
+          LET tagsOut = collected[*].tg
+          // limit
+          LIMIT @offset, @limit
+          // respond
+          RETURN { dit: ${ditType}Out, tags: tagsOut }`;
+    const params = { tagnames, offset, limit, '@ditTags': ditTags };
+    const cursor = await this.db.query(query, params);
+    const out = await cursor.all();
+
+    // generate dit-tags ids, and add them as attributes to each dit
+    // and return array of the dits
+    return out.map(({ dit, tags }) => {
+      dit[`${ditType}Tags`] = tags.map(({ tagname }) => ({
+        id: `${dit.id}--${tagname}`,
+        [`${ditType}`]: dit,
+        tag: { tagname }
+      }));
+      return dit;
+    });
+  }
+
+  /**
+   * Read new dits
+   */
+  static async findNew(ditType, { offset, limit }) {
+    const ditCollection = ditType + 's';
+    const query = `
+      FOR ${ditType} IN @@ditCollection
+        // sort from newest
+        SORT ${ditType}.created DESC
+        LIMIT @offset, @limit
+        // find creator
+        LET c = (DOCUMENT(${ditType}.creator))
+        LET creator = MERGE(KEEP(c, 'username'), c.profile)
+        // format for output
+        LET ditOut = MERGE(KEEP(${ditType}, 'title', 'detail', 'created'), { id: ${ditType}._key}, { creator })
+        // limit
+        // respond
+        RETURN ditOut`;
+    const params = { offset, limit, '@ditCollection': ditCollection };
+    const cursor = await this.db.query(query, params);
+    return await cursor.all();
+  }
+
+  /**
+   * Read random dits
+   * @param {number} [limit] - max amount of random dits to return
+   */
+  static async random(ditType, { limit }) {
+    const ditCollection = ditType + 's';
+    const query = `
+      FOR ${ditType} IN @@ditCollection
```

Comment:
is it necessary to make the query (and other queries) variable like this?
Perhaps it would be more straightforward and secure to use `dit` instead of `${ditType}` in this query?